### DISCUSSION TITLE
Don't dismiss Connection Setup dialog after connect

### DISCUSSION THREAD

**Discussion by chrisrimple** [jamulussoftware/jamulus#1372] [Ideas]
In 3.7.0, opening the Connection Setup dialog and then connecting to a server causes the Connection Setup dialog to be dismissed.  There are situations where it would be great if the Connection Setup dialog could remain open, making it easier to then switch to another server.  This would be particularly beneficial for the volunteers and performers on the WorldJam.

Don't change the default behavior (dismiss the Connection Setup dialog), but allow it to be changed via a (new) checkbox in the Connection Setup dialog OR a new menu item on the Edit menu.  Perhaps not necessary to save the setting between sessions.

**Comment by gilgongo** [comment]
I believe the reason it closes is to minimise the considerable traffic that it generates which could impact the stability of a jam session. It has been suggested (on the UI thread I think) that to allow people to leave it open it should only poll once or twice when opened, and then have a "refresh" button for this reason.

Either way, so that we can have agreed work tickets on the issues list, I'll move this to discussion prior to a spec for any resulting PR.

**Reply by dcorson-ticino-com** [reply]
You can open the connection dialog again without disconnecting using View>Connection Setup.
This does, however, add to the network traffic.

**Reply by chrisrimple** [reply]
@gilgongo makes a good point about the network traffic.  For this use case, a static list with a Refresh button would be fine.

@dcorson-ticino-com I'm aware that the Connection Setup dialog can be opened again, but that's the point: during a WorldJam, some of the backstage volunteers are changing servers literally 50-100 times.  So the goal is to make it easier/faster to switch between servers.

As an alternative to leaving it open, a Favorites List of servers (IP addresses) would allow WorldJam volunteers to quickly switch servers without the need to reopen the Connection Setup dialog, so no network traffic.

**Reply by dcorson-ticino-com** [reply]
If you really what to try a version with continuously open connect you can download my new UI beta:
https://github.com/dcorson-ticino-com/jamulus/releases/tag/r3_8_0beta3NewUI

This is a completely new UI with tabbed docked windows.  The connect window is updated whenever it is opened, a cycle of 8 pings every 2.5 seconds is done.  After that it pings all the servers once a minute or the cycle can be restarted by clicking update.

Note that this is a quick hack and does not have all the features like keyboard shortcuts etc.
Please let me know your experience with this version.

**Reply by gene96817** [reply]
How much traffic does the open connection window generate?
I like having the connection window open when my chorus has sectionals. Then I can see the attendance at the parallel sessions. The updates don't have to be very frequent (once a minute seems fine). In a second use case, yet related, when I am practicing online, I like to monitor some far-away servers (> 40ms away) to visit friends. An open connection window is the only method. A low ping rate is sufficient (and a refresh button would be an annoyance).

**Reply by dcorson-ticino-com** [reply]
>How much traffic does the open connection window generate?

I think that if you hear no perturbations of the sound then you are OK.
I measured it once, but don't remember exactly.
If you are already a boarder case it could be enough to cause problems.

Try my NewUI, link above.  After the first 20 seconds the connect window updates once a minute.
You can undock it and leave it open forever. A double-click disconnects and connects to the new server.
